### PR TITLE
Pull libfprint autosuspend hwdb

### PR DESCRIPTION
**Pull request by benzea**

### CODE REVIEW

File: tools/autosuspend-update.sh [hunk 1]
```diff
@@ -4,4 +4,12 @@ set -eu
...
 (curl --fail -L 'https://chromium.googlesource.com/chromiumos/platform2/+/master/power_manager/udev/gen_autosuspend_rules.py?format=TEXT'; echo) \
-    | base64 -d > gen_autosuspend_rules.py
+    | base64 -d > tools/chromiumos/gen_autosuspend_rules.py
+
+(cat <<%EOF
+# This file is part of systemd.
+#
+# Pulled from libfprint upstream in order to autosuspend fingerprint readers
+%EOF
+curl -L 'https://gitlab.freedesktop.org/libfprint/libfprint/-/jobs/artifacts/master/raw/60-autosuspend-libfprint.hwdb?job=build') \
```
Comment (bluca) [inline tools/autosuspend-update.sh]: please use curl --fail as above

File: tools/autosuspend-update.sh [hunk 2]
```diff
@@ -4,4 +4,12 @@ set -eu
 cd "$1"
 
 (curl --fail -L 'https://chromium.googlesource.com/chromiumos/platform2/+/master/power_manager/udev/gen_autosuspend_rules.py?format=TEXT'; echo) \
-    | base64 -d > gen_autosuspend_rules.py
+    | base64 -d > tools/chromiumos/gen_autosuspend_rules.py
+
+(cat <<%EOF
+# This file is part of systemd.
+#
+# Pulled from libfprint upstream in order to autosuspend fingerprint readers
```
Comment (poettering) [inline tools/autosuspend-update.sh]: should probably mention the license/copyright holder of the source of this in some way. better mention this too much then too little.

File: hwdb.d/meson.build [hunk 3]
```diff
@@ -20,6 +20,7 @@ hwdb_files_notest = files('''
 
 hwdb_files_test = files('''
         60-autosuspend.hwdb
+        60-autosuspend-libfprint.hwdb
```
Comment (poettering) [inline hwdb.d/meson.build]: hmm, ideally the hwdb files would be named after contents not their source.

File: tools/autosuspend-update.sh (hunk 2 above)
Comment (benzea) [reply tools/autosuspend-update.sh]: Good point. We shuffled this around upstream and also added a SPDX license header there as that seemed more sensible than adding it on the systemd side.

**Review by bluca — APPROVED** [review]
Looks fine and comments all applied, setting green label

**Comment by bluca** [comment]
> Looks fine and comments all applied, setting green label

But waiting for @keszybz to have a look as well

File: tools/autosuspend-update.sh [hunk 4]
```diff
@@ -4,4 +4,4 @@ set -eu
 cd "$1"
 
 (curl --fail -L 'https://chromium.googlesource.com/chromiumos/platform2/+/master/power_manager/udev/gen_autosuspend_rules.py?format=TEXT'; echo) \
-    | base64 -d > gen_autosuspend_rules.py
+    | base64 -d > tools/chromiumos/gen_autosuspend_rules.py
```
Comment (keszybz) [inline tools/autosuspend-update.sh]: ~~Please restore EOL.~~

I see this happens in patch 2.

**Review by keszybz — APPROVED** [review]
LGTM.

**Pull request merged**